packages/frontend/apps/electron/src/main/protocol.ts

```diff
@@ -41,7 +42,13 @@ function isNetworkResource(pathname: string) {
 }
 
 async function handleFileRequest(request: Request) {
-  const urlObject = new URL(request.url);
+  let url = request.url;
+
+  if (url.startsWith(anotherOrigin)) {
+    url = url.replace(anotherOrigin, mainWindowOrigin);
+  }
+
+  const urlObject = new URL(url);
```

Comment:
you can directly do it with `new URL(request.url); url.host = mainHost`. it's faster